### PR TITLE
Added Merge Sort [Java] 

### LINKED ISSUE #61: Merge Sort
https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort

### PR DESCRIPTION
**Pull request by jsroyal**
Fixed #61 

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

**Comment by jsroyal** [comment]
Oh, Sorry. Again added this file.
@singhpratyush What is the way to remove please can you help?

**Comment by singhpratyush** [comment]
Create a new branch from the `upstream/master` and make changes there.

**Comment by jsroyal** [comment]
Do I need to new pull request  or same ? So please?

**Comment by singhpratyush** [comment]
You may continue working on the same pull request. But you will have to remove the extra file and sync with master.

**Comment by jsroyal** [comment]
I created new branch and syn with upstream/master and when I push it say new pull request?

**Comment by singhpratyush** [comment]
I'm sorry but I didn't get that

**Comment by jsroyal** [comment]
I created new branch and set remote upstream/master and syn and put file merge sort.
Is that that to resolve ?

**Comment by singhpratyush** [comment]
If you wish to continue with this PR, you will have to merge `upstream/master` with this branch itself.

**Comment by jsroyal** [comment]
git branch --set-upstream-to=origin/master Merge_Sort
  466  git pull
  467  git status
  468  git add .
  469  git commit -m "added merge sort"
  470  git branch
  471  git push
  472  git push origin Merge_Sort 
@singhpratyush Please check out ?

**Comment by singhpratyush** [comment]
That would create the branch `Merge_Sort`, but this one is `merge_sort`.

PS: Please stop creating branches in this repository. Do it on your version of the repository.

**Comment by jsroyal** [comment]
So what could be better way please I'm confused?
PS: Sorry for creating two branch but to better way please :)

**Comment by jsroyal** [comment]
I want to send in same PR.

**Comment by singhpratyush** [comment]
This PR is linked with the branch `merge_sort`. Work on the same branch to reflect changes here. Try merging `Merge_Sort` with `merge_sort`.

**Comment by jsroyal** [comment]
@singhpratyush Please check it out :)

File: README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Community (college) maintained list of algorithm implementations and codes.
 | [Longest Common Subsequence](http://www.geeksforgeeks.org/dynamic-programming-set-4-longest-common-subsequence) | [\[X\]](longest_common_subsequence/longestCommonSubsequence.c) | | | |
 | [Quick Select](https://en.wikipedia.org/wiki/Quickselect) | [\[X\]](quick_select/quick_select.c) | | | |
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | | |
+| [Merge Sort](https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort) | | | [\[X\]](merge_sort/MergeSort.java) | |
```
Comment (Monal5031) [inline README.md]: We are adding the list alphabetically, so please adjust it accordingly

Comment (jsroyal) [inline README.md]: Okay.

File: README.md [hunk 2]
```diff
@@ -17,9 +17,11 @@ Community (college) maintained list of algorithm implementations and codes.
 | [Linear Search](https://en.wikipedia.org/wiki/Linear_search) | | | | [\[X\]](linear_search/linear_search.py) |
 | [Longest Common Subsequence](http://www.geeksforgeeks.org/dynamic-programming-set-4-longest-common-subsequence) | [\[X\]](longest_common_subsequence/longestCommonSubsequence.c) | | | |
 | [Quick Select](https://en.wikipedia.org/wiki/Quickselect) | [\[X\]](quick_select/quick_select.c) | | | |
+| [Merge Sort](https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort) | | | [\[X\]](merge_sort/MergeSort.java) | |
```
Comment (Monal5031) [inline README.md]: It will come before quick select.

File: README.md [hunk 3]
```diff
@@ -17,9 +17,11 @@ Community (college) maintained list of algorithm implementations and codes.
...
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | | |
 
 
+
```
Comment (Monal5031) [inline README.md]: Remove this extra line too

File: README.md (hunk 2 above)
Comment (Monal5031) [inline README.md]: Please correct this one too.

**Pull request closed without merging**